docs/io/visualization/using_widgets.rst

```diff
@@ -178,4 +178,37 @@ There are also several other options in the modebar which we have not explained
 you remember to click back on the **Box Select** option for making selections on
 spectrum.
 
+Energy Level Widget
+################
+
+This widget lets you visualize the last line interactions
+
+.. image:: ../images/energy_level_widget_demo.gif
+    :alt: Demo of Energy Level Widget
+
+By selecting an ion on the widget, you can see its energy level diagram based
+on the last line interactions in the simulation.
+
+The y-axis of the plot represent energies and the horizontal levels represent the
+different energy levels. The width of these platform represent the level populations
+in those levels.
+
+The arrows represent the line interaction, with the upward arrows
+represent excitation and vice versa. The width of the arrows represents the number
+of packets which underwent that interaction, while the color of the arrows represent
+the wavelength of the transition.
```

Comment:
Arrows represent the line interactions experienced by packets. Upwards arrows show excitation from lower energy levels to higher levels and downward arrows show de-excitation from higher energy levels to lower levels. The thickness of each arrow represents the number of packets that underwent that interaction. The wavelength of the transition is given by the color.